client/ayon_core/hosts/maya/plugins/workfile_build/script_placeholder.py

```diff
@@ -0,0 +1,201 @@
+
+    This is a very low-level placeholder to run Python scripts at a given
+    point in time during the workfile template build.
+
+    It can create either a locator or an objectSet as placeholder node.
+    It defaults to an objectSet, since allowing to run on e.g. other
+    placeholder node members can be useful, e.g. using:
+
+    >>> members = cmds.sets(placeholder.scene_identifier, query=True)
+
+    """
+
+    identifier = "maya.runscript"
+    label = "Run Python Script"
+
+    use_selection_as_parent = False
+
+    def get_placeholder_options(self, options=None):
+        options = options or {}
+        return [
+            NumberDef(
+                "order",
+                label="Order",
+                default=options.get("order") or 0,
+                decimals=0,
+                minimum=0,
+                maximum=999,
+                tooltip=(
+                    "Order"
+                    "\nOrder defines asset loading priority (0 to 999)"
+                    "\nPriority rule is : \"lowest is first to load\"."
+                )
+            ),
+            TextDef(
+                "prepare_script",
+                label="Run at\nprepare",
+                tooltip="Run before populate at prepare order",
+                multiline=True,
+                default=options.get("prepare_script", "")
+            ),
+            TextDef(
+                "populate_script",
+                label="Run at\npopulate",
+                tooltip="Run script at populate node order<br>"
+                        "This is the <b>default</b> behavior",
+                multiline=True,
+                default=options.get("populate_script", EXAMPLE_SCRIPT)
+            ),
+            TextDef(
+                "depth_processed_script",
+                label="Run after\ndepth\niteration",
+                tooltip="Run script after every build depth iteration",
+                multiline=True,
+                default=options.get("depth_processed_script", "")
+            ),
+            TextDef(
+                "finished_script",
+                label="Run after\nbuild",
+                tooltip=(
+                    "Run script at build finished.<br>"
+                    "<b>Note</b>: this even runs if other placeholders had "
+                    "errors during the build"
+                ),
+                multiline=True,
+                default=options.get("finished_script", "")
+            ),
+            EnumDef(
+                "create_nodetype",
+                label="Nodetype",
+                items={
+                    "spaceLocator": "Locator",
+                    "objectSet": "ObjectSet"
+                },
+                tooltip=(
+                    "The placeholder's node type to be created.<br>"
+                    "<b>Note</b> this only works on create, not on update"
+                ),
+                default=options.get("create_nodetype", "objectSet")
+            ),
```

Comment:
What is the use case for having 2 types of nodes for this?

Reply:
I have the same question of which toke has asked. 
And also shouldn't we add the `multiselection=True` into Enum Definition when there are multiple options being used or the related function has been updated to allow this happened without the argument?

Reply:
> What is the use case for having 2 types of nodes for this?

It's so that the placeholder can be either a locator or an objectSet. It being an object set, may mean that it can e.g. contain multiple placeholders for the 'script' to act upon. It being a locator it can be used to e.g. specify a particular point in a hierarchy or whatever. Both may have their own reasons to be used.

A locator may also be more valuable in a 'nested/recursive' load of placeholders since the loaded hierarchy containing a placeholder of its own might already be parented into another hierarchy not known upon creation of the placeholder, etc.

> And also shouldn't we add the `multiselection=True` into Enum Definition when there are multiple options being used or the related function has been updated to allow this happened without the argument?

Not sure what you mean. It can be **either** an object set or a locator - not both at the same time.

Reply:
> Not sure what you mean. It can be either an object set or a locator - not both at the same time.

I think I misunderstood the code a little bit. 
I get what you mean when you said the default value can be either an object set or a locator.

Reply:
So if a look locator is a child of another placeholder, the look will be assigned to its loaded parent container?

Reply:
> So if a look locator is a child of another placeholder, the look will be assigned to its loaded parent container?

This particular PR is unrelated to looks. It's a placeholder that runs custom Python scripts defined by the user on creation.

Reply:
Sorry, I need to be awake before I go on github.

Reply:
No worries - I'll mark this conversation as resolved. Let me know if there are other issues! :)